src/js/components/SidebarVolumesFilterComponent.jsx

```diff
@@ -0,0 +1,74 @@
+import React from "react/addons";
+
+import AppsStore from "../stores/AppsStore";
+import AppsEvents from "../events/AppsEvents";
+import FilterTypes from "../constants/FilterTypes";
+
+import QueryParamsMixin from "../mixins/QueryParamsMixin";
+
+/**
+ * Health is wealth, peace of mind is happiness.
+ */
+var SidebarVolumesFilterComponent = React.createClass({
+  displayName: "SidebarVolumesFilterComponent",
+
+  mixins: [QueryParamsMixin],
+
+  handleChange: function () {
+    var currentState = this.getQueryParamValue(FilterTypes.VOLUMES) === "true"
+      ? null
+      : true;
+
+    this.setQueryParam(FilterTypes.VOLUMES, currentState);
+  },
+
+  getInitialState: function () {
+    return {
+      appsVolumesCount: ""
+    };
+  },
+
+  componentWillMount: function () {
+    AppsStore.on(AppsEvents.UPDATE_APPS_FILTER_COUNT,
```

Comment:
It could have been done with the `CHANGE` event but then I do have to cycle *again* over the applications and count the ones with volumes, which is currently done directly with the updating of the health filter and the status filter. Also the labels filter works different from the volumes filter volumes is a simple counter rather than a list of all available options like the labels filter. Which is triggered by the `CHANGE` event in the `AppListComponent`. Should we change that behaviour?